### PR TITLE
[WIP] Apple pay inbox notification

### LINKED ISSUE #5099: New inbox notification: Apple Pay
**Title:** Apple Pay is now available on Stripe
**Content:** Boost sales by offering a fast, simple, and secure checkout experience with Apple Pay®!
**Segmentation:** Stores located in the US with Stripe installed and activated
**Timing:** When users update WooCommerce to the 4.7 version (Nov 10)
**Cadence:** Once
**CTA leads to:** https://woocommerce.com/apple-pay/?utm_source=inbox
**CTA label:** Get started
**Type:** Marketing/promotion

More context: p7fD6U-3RJ-p2

### PR DESCRIPTION
**Pull request by becdetat**
Fixes #5099

WIP, waiting on final CTA link from @pmcpinto 

Adds a notification that Apple Pay is now available on Stripe.

### Screenshots

Screen Shot 2020-09-25 at 5 08 19 pm

### Detailed test instructions:

- Set the site to a US location.
- Install and activate the [WooCommerce Stripe Payment Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/)
- Go to the WooCommerce home screen.
- Ensure the `Apple Pay is now available on Stripe` is displayed in the inbox.

### Changelog Note:

Tweak: Inbox notification added for Apple Pay on Stripe.

### CODE REVIEW

File: src/Notes/ApplePayTBD.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	 * @return Note|null
+	 */
+	public static function get_note() {
+		// Only show if Stripe plugin is active.
+		if ( ! PluginsHelper::is_plugin_active( 'woocommerce-gateway-stripe/woocommerce-gateway-stripe.php' ) ) {
+			return;
+		}
+
+		// Only show if in the US.
+		$base_location = wc_get_base_location();
```
Comment (becdetat) [inline src/Notes/ApplePayTBD.php:40]: I did have some nice array destructuring here but key-based destructuring is only supported in PHP 7.2

**Comment by pmcpinto** [comment]
@becdetat @adrianduffell just a heads up that this one should only be included in 1.7 (WC 4.7). We'll probably need to make some changes in the CTA link

**Review by joshuatf — APPROVED** [review]
This is testing well for me after dumping autoload files and running `wc_admin_daily` cron event.  Love the simplification of class names here.

Pre-approving pending a couple changes, but I think the deprecation comments can be ignored if we plan to follow-up in https://github.com/woocommerce/woocommerce-admin/pull/5159

File: src/Notes/DeprecatedNotes.php [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+ * Define deprecated classes to support changing the naming convention of
+ * admin notes.
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+
+/**
+ * WC_Admin_Note (deprecated, use Note instead.
```
[the remark this answers is not in the document]
Comment (becdetat) [reply src/Notes/DeprecatedNotes.php]: This was fixed in the main PR, I've merged in master and that's cleared up this PR. I'll add the `@deprecated` tag in a seperate PR, thanks!

File: src/Notes/DeprecatedNotes.php [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+ */
+class WC_Admin_Note extends Note {}
```
[the remark this answers is not in the document]
Comment (becdetat) [reply src/Notes/DeprecatedNotes.php]: Yes, that's merged in `main`

File: src/Notes/ApplePayStripe.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		}
+		if ( 'US' !== $base_location['country'] ) {
+			return;
+		}
+
+		$note = new Note();
+
+		$note->set_title( __( 'Apple Pay is now available on Stripe', 'woocommerce-admin' ) );
+		$note->set_content( __( 'Boost sales by offering a fast, simple, and secure checkout experience with Apple Pay®!', 'woocommerce-admin' ) );
+		$note->set_content_data( (object) array() );
```
[the remark this answers is not in the document]
Comment (becdetat) [reply src/Notes/ApplePayStripe.php:52]: Copy-pasta, most other note factories have this line. FWIR it's not necessary. I'll take it out.

**Review by joshuatf — APPROVED** [review]
Thanks for the changes, Bec!  LGTM 🚢 :shipit: 🚀

**Comment by becdetat** [comment]
Note that there is discussion around this - it will probably be changed significantly before going live - paqDfi-HW-p2

**Comment by dechov** [comment]
@becdetat Following a subthread in that discussion (paqDfi-HW-p2#comment-2506), I've done a test (https://github.com/woocommerce/woocommerce-gateway-stripe/commit/693adbaa7a2b469d3ca2df725c6b3bd11896d9c8) of adding a note directly from the Stripe extension (based on this PR), which sounds like it may be best for any notes limited to an audience of those with the extension already installed – let me know if there are any concerns I may be overlooking 🙂

**Comment by becdetat** [comment]
@dechov sorry, this escaped my attention :-)

> let me know if there are any concerns I may be overlooking

I've added a couple of notes but I can't see any issues, it all looks good to me.

**Comment by dechov** [comment]
I believe this can be closed, as it has now been implemented between https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1337 and https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1365. Thanks for getting this started!

**Comment by becdetat** [comment]
@dechov Thanks for letting us know :-) closing this now

**Pull request closed without merging**